### PR TITLE
Add setup logging to cli, ui, ekuiper, edgexfoundry tests

### PR DESCRIPTION
**Pull request by MonicaisHer**

### CODE REVIEW

File: test/suites/ekuiper/main_test.go [hunk 1]
```diff
@@ -24,51 +24,26 @@ const (
...
-		goto TEARDOWN
+
+	code := m.Run()
+	if code != 0 {
+		teardown()
+		os.Exit(code)
 	}
 
 	// security off
+	testSecretsInterface = false
```
Comment (farshidtz) [inline test/suites/ekuiper/main_test.go:45]: I suggest printing a log message here to show we are repeating the tests with disabled security.

File: test/suites/cli/main_test.go [hunk 2]
```diff
@@ -5,38 +5,56 @@ import (
...
+	utils.SnapRemove(nil,
+		snapName,
+	)
+
+	log.Println("[SETUP]")
+	start := time.Now()
+
+	teardown = func() {
+		log.Println("[TEARDOWN]")
+		utils.SnapDumpLogs(nil, start, snapName)
```
Comment (farshidtz) [inline test/suites/cli/main_test.go:42]: CLI has no service, so no logs.

**Review by farshidtz — CHANGES_REQUESTED** [review]

**Review by farshidtz — APPROVED** [review]
Thanks for the changed @MonicaisHer.

File: test/suites/ekuiper/main_test.go [hunk 3]
```diff
@@ -81,23 +57,8 @@ func TestMain(m *testing.M) {
...
-	utils.SnapRemove(nil,
-		ekuiperSnap,
-		"edgexfoundry",
-		deviceVirtualSnap,
-		ascSnap,
-	)
-
-	os.Exit(exitCode)
+	code = m.Run()
+	os.Exit(code)
```
Comment (farshidtz) [inline test/suites/ekuiper/main_test.go:61]: @MonicaisHer Teardown is not called here, so the snaps aren't removed and logs aren't collected.

Comment (MonicaisHer) [reply test/suites/ekuiper/main_test.go:61]: `teardown()` is missed here

Comment (MonicaisHer) [reply test/suites/ekuiper/main_test.go:61]: this will be solved by https://github.com/canonical/edgex-snap-testing/pull/115

**Pull request merged**